pkg/kubelet/network/dns/dns.go

```diff
@@ -67,37 +78,34 @@ func NewConfigurer(recorder record.EventRecorder, nodeRef *v1.ObjectReference, n
 	}
 }
 
-func omitDuplicates(pod *v1.Pod, combinedSearch []string) []string {
-	uniqueDomains := map[string]bool{}
+func omitDuplicates(strs []string) []string {
+	uniqueStr := make(map[string]bool)
```

Comment:
Can we just use the k8s `sets` data structure?

Reply:
Maybe keep this for now? There are some existing unit tests written in a bad way that they assume strings order is preserved. Trying to not touch them at this point so we know previous behavior is preserved. Would like to clean them up after.

Reply:
oh, I see what this is doing. Can we rewrite it in a "less efficient" but more obvious way?
```
var ret []string
for _, str := range strs {
  if ! uniqueStr[str] {
    ret = ret.append(str)
    uniqueStrs[str] = true
  }
}
return ret
```

Reply:
Sounds good, suggestion applied.